### ISSUE TITLE
Simpler Processor and Exporter Configuration

### ISSUE THREAD

**Issue by tsloughter** [open-telemetry/opentelemetry-erlang#345]
Configuration of the processor and exporter looking like:

```
{opentelemetry,
  [{processors, 
    [{otel_batch_processor,
        #{exporter => {opentelemetry_exporter, #{protocol => grpc,
                                                 endpoints => ["http://localhost:9090"],
                                                 headers => [{"x-honeycomb-dataset", "experiments"}]}}}}]}]}
```

Is sort of for historical reasons. The way processors evolved in the spec made this really not worth the deep configuration to support.

Basically processors are more limited and the spec added a line about "decorators" that can be added to a processor like the batch processor to do additional work -- what I originally thought you would use multiple processors for.

So while I always thought we should keep this existing configuration as an option, since it is technically correct and maybe will be useful again in the future (and of course since we've released 1.0.0 we have to keep it anyway :) I think a simpler configuration could be done:

```
[
 {opentelemetry,
  [{span_processor, batch},
   {exporter, otlp}]},

 {opentelemetry_exporter, 
    [{protocol, grpc},
     {endpoints, ["http://localhost:9090"]}
     {headers, [{"x-honeycomb-dataset", "experiments"}]}]}
].
```

``` elixir
config :opentelemetry, 
    span_processor: :batch,
    exporter: otlp
    
config :opentelemetry_exporter,
    protocol: :grpc
    endpoints: ["http://localhost:9090"],
    headers: [{"x-honeycomb-dataset", "experiments"}]
```

Should be able to configure the exporter in `opentelemtry` configuration and not only `opentelemetry_exporter`, I'm just not sure what that would best look like yet.

**Comment by tsloughter** [comment]
Just realized it needs to be something like `span_exporter` since metrics exporter will also exist, and possibly other signal exporters.

**Comment by tsloughter** [comment]
Resolved by #357
